bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MANUFACTURER, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, device.getModel());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_MODEL, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, device.getFullDescription());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve model information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_COMBINED_ID, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, device.getPJLinkClass());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_CLASS, e.toString());
+        }
+        try {
+            Map<ErrorStatusDevicePart, ErrorStatusQueryResponseState> errorStatus = device.getErrorStatus();
+            for (Map.Entry<ErrorStatusDevicePart, ErrorStatusQueryResponseState> entry : errorStatus.entrySet()) {
+                String key = entry.getKey().getCamelCaseText();
+                String value = entry.getValue().getText();
+                updateProperty(PJLinkDeviceBindingConstants.PARAMETER_ERROR_STATUS + key, value);
+            }
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_ERROR_STATUS, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_LAMP_HOURS, device.getLampHours());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_LAMP_HOURS, e.toString());
+        }
+        try {
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_OTHER_INFORMATION, device.getOtherInformation());
+        } catch (ResponseException e) {
+            // okay, cannot retrieve class information
+            updateProperty(PJLinkDeviceBindingConstants.PARAMETER_OTHER_INFORMATION, e.toString());
+        }
+    }
+
+    private void updateInputChannelStates(PJLinkDevice device)
+            throws ResponseException, IOException, AuthenticationException {
+
+        Set<Input> inputs = device.getAvailableInputs();
+        // add our activities as channel state options
+        List<StateOption> states = new LinkedList<>();
+        for (Input input : inputs) {
+            states.add(new StateOption(input.getPJLinkRepresentation(), input.getText()));
+        }
+
+        ChannelTypeUID channelTypeUID = new ChannelTypeUID(
```

Comment:
You can totally replace all this by providing a DynamicStateDescriptionProvider.
And you don't need two different channels, inputChannel and inputChannelDynamic. inputChannel with dynamic options will be sufficient.

Reply:
done